### DISCUSSION TITLE
nanoCLR task not start correctly on ESP32-S3!

### DISCUSSION THREAD

**Discussion by omidg2** [nanoframework/Home#1536] [Q&A]
### Target name(s)

ESP32-S3

### Description

 I use ESP32-S3, I flash nanoframework image to it by this command line:
nanoff --masserase --update --target ESP32_S3_BLE --serialport COM5
the nanoframework image was program on ESP32-S3 correctly as you can see on the first attached screen shot...

but when I connect my device to putty with baudrate 115200 and start it, it seems nanoCLR not start correctly, as you can see
in second screen shot

I use nanoFramework on other version of ESP32 like ESP32-WROOM-32 and it works correctly

### Expected behaviour

nanoCLR task start correctly and we can see the device in Visual Studio Device Explorer...

### Screenshots

![1](https://github.com/user-attachments/assets/06ac5d45-4996-4c2e-99b4-388a86d4190d)
![2](https://github.com/user-attachments/assets/652180fa-694f-4ea3-b6ce-a1fdf4e38b30)

**Comment by alberk8** [comment]
Which  ESP32 S3 do you have? From which manufacturer, make and model?

**Comment by omidg2** [comment]
> Which ESP32 S3 do you have? From which manufacturer, make and model?

This is full part number:  ESP32-S3-WROOM-1U-N4
as I mentioned, nanoFramework Image pragrammed correctly, (ESP32_S3_BLE 1.12.0.47)
but device not found in Visual Studio 22 Device Explorer, I check the problem and it seems nanoCLR task not start....

**Comment by alberk8** [comment]
Looks good and based on the screen capture it looks like the boot process is good.  Are you sure that there are no other serial terminal app is opened when VS 2022 you plug in the ESP32 S3 and view it in Device Explorer?  Make sure that the Device Watcher is **enabled** in Device Explorer.

**Comment by omidg2** [comment]
No, not good, if you see second screen capture, you can see that nanoCLR not started correctly, it stops in line
**_I (304) boot: Disabling RNG early entropy source ..._**
a good start up should be something like this screen capture:
![3](https://github.com/user-attachments/assets/c31f285f-9603-4868-8295-057e91e9dd2d)
this screen capture is from another board which works correctly, 
as you see here, many lines such as 
**_I (415) cpu_start: Multicore app_**

and lines after that should be exist,

**Comment by alberk8** [comment]
If your S3 has 2 usb-c port then VS will only detect it on the other USB port (usually the left hand side usb c).

**Comment by omidg2** [comment]
Dear Alber, I do not use USB, I use USART 0 on pin 36 and  pin 37  of module, as you can see in the picture below:
![image](https://github.com/user-attachments/assets/0f9b79e8-2cad-4273-87cd-6a6027482143)
my communications to module, fully are done through this USART port ...

**Comment by alberk8** [comment]
The current S3 firmware expect to communicated with VS 2022 via USB CDC, you could [compile ](https://github.com/nanoframework/nf-interpreter/blob/b777c0ee29475c6a53cbedb07604e5ed116370b5/targets/ESP32/CMakePresets.json#L358) firmware with CDC off.

**Comment by omidg2** [comment]
It means if I connect my PC via USB port to IO19 and IO20 of module, like below diagram, it might work properly?
![image](https://github.com/user-attachments/assets/54e6fce8-777a-4342-93fe-6ca1a70daae1)

**Comment by alberk8** [comment]
most likely will work

**Comment by omidg2** [comment]
Thank you alberk8, I tested it and it didn't work, I'm probably doing something wrong somewhere,

**Reply by alberk8** [reply]
Check if you have the D+ and D- connected properly.

**Reply by omidg2** [reply]
I checked it ...the connection is OK, but not detected by windows...

**Comment by belem2050** [comment]
@josesimoes , just to say, it looks like we're having trouble with the latest image release today.

``
18:29:11:370	ESP32_S3 @ COM13 is active running nanoCLR.
18:29:13:564	18:29:13.556 [Starting deployment transaction from v2022.12.1.22+02c1ad45ca]
18:29:13:565	18:29:13.556 [Try to get exclusive access to the nanoDevice]
18:29:13:565	18:29:13.556 [Starting debug engine on nanoDevice]
18:29:14:011	OK]
18:29:14:068	18:29:14.026 [nanoDevice is not STM32, skipping flash dump]
18:29:15:522	18:29:15.507 [Version mismatch for mscorlib. Need v1.15.6.0, checksum 0x445C7AF9.]
18:29:15:522	18:29:15.507 [The connected target has v100.5.0.19, checksum 0xC5D5F755.]
18:29:19:191	18:29:19.188 [NanoDevices: device departure COM12]
``.


I am not exactly sure from where it is from but going to the previous ESP32_S3 firmware version, Ifixed the issue.

**Comment by josesimoes** [comment]
@belem2050 https://discord.com/channels/478725473862549535/481780524864503828/1334507682152841278
